include/gl_depth_sim/interfaces/noise_model.h

```diff
@@ -0,0 +1,17 @@
+#ifndef NOISE_MODEL_H
+#define NOISE_MODEL_H
+
+#include <math.h>
+#include <vector>
+#include <iostream>
+#include <iomanip>
+#include <string>
+#include <map>
+#include <random>
+
+namespace gl_depth_sim
+{
+std::vector<float> noise(std::vector<float> distance);
```

Comment:
Passing and returning very large objects (like these vectors) can be very expensive. Consider passing the input as a constant reference: `const std::vector<float>& distance`.

There's arguments to be made that returning by value is not so bad, but you do have options:
1. Return with an output parameter (so that memory can be re-used)
```c++
void noise(const std::vector<float>& distance_in,  std::vector<float>& distance_out);
```
2. Modify the vector "in-place":
```c++
void (std::vector<float>& distance);
```